### DISCUSSION TITLE
MetaData mixing up parent and child tables?

### DISCUSSION THREAD

**Discussion by Wout45** [sqlalchemy/sqlalchemy#11784] [Usage Questions] [answered]
I have 2 tables:

table1: id_unique, description (only id_unique is PK)
table2: id, location (both columns form a combined PK)

The relationship between the 2 tables goes as following: 

table2 has a foreign Key on 'id' referencing table1 column 'id_unique'. To put it simple, I allow table2 to contain the same 'id' multiple times as long as the 'location' is different.
I have rows in table2 with the same 'id' value multiple times and only unique 'id_unique' values in table1 which verify that the relationship is setup correctly.

So now I use the sqlAlchemy function 'MetaData' to extract all properties of table2.
It can correctly identify that it has a foreign key which is called "ForeignKey('table1.id_unique')". However, when I try to get the specific details of that foreign key, it gets all mixed up and I don't understand why.

This is the debugger output it get from the following functions used on the foreign key of Table2:

foreign_key.parent.table.name = 'table2'
foreign_key.parent.name = 'id'

foreign_key.column.table.name = 'table1'
foreign_key.column.name = 'id_unique'

I've always been under the impression that the parent table holds the unique identifier of the relationship, however here it seems to be reversed and I don't understand why.

I've read through the documentation, where the unique identifier table is also considered as the parent table, so I'm a bit lost now. Technically my code runs fine if I just keep it in my mind that I have to switch parent and child, but it doesn't feel right that I have to do it like this to get the correct result, there must be a reason why it's switched.

Thanks for your help! :)

**Comment by gordthompson** [comment, marked as answer]
> my code runs fine if I just keep it in my mind that I have to switch parent and child

If your code is digging into the internals of a ForeignKey object to glean information about the FK's properties you may find it more straightforward to use the `inspect()` interface

```python
insp = inspect(engine)
pprint(insp.get_foreign_keys("table2"), sort_dicts=False)
"""
[{'name': 'FK__table2__id__2A6B46EF',
  'constrained_columns': ['id'],
  'referred_schema': None,
  'referred_table': 'table1',
  'referred_columns': ['id_unique'],
  'options': {}}]
"""
```

As to why the name `parent` was used internally, that's *way* before my time, but it might be "parent" in terms of the relationship between objects in the class, rather than "parent" in terms of the FK (parent-child) relationship between the tables.

**Reply by Wout45** [reply]
I've tried inspect() and the outputs are exactly what I expect.
Thank you for your time and help!
